File: pbgestfinbowcl/src/app/gestione-crediti/services/revoca-bancaria.service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { HttpClient, HttpClientModule, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { ConfigService } from "src/app/core/services/config.service";
import { HandleExceptionService } from "src/app/core/services/handle-exception.service";
import { RevocaBancariaDTO } from "../commons/dto/revoca-bancaria-dto";

import { StoricoRevocaDTO } from "../commons/dto/storico-revoca-dto";

@Injectable()
export class RevocaBancariaService {
  
    
    getRevoca(idUtente: number, idProgetto: number, idModalitaAgevolazione:number): Observable <RevocaBancariaDTO>  {
      let url = this.configService.getApiURL() + '/restfacade/revoca/getRevocaBancaria';
       let params = new HttpParams()
      .set("idUtente", idUtente.toString())
      .set("idProgetto", idProgetto.toString()).set("idModalitaAgevolazione", idModalitaAgevolazione.toString());
       return this.http.get<RevocaBancariaDTO>(url, { params: params });
    }

    
    salvaRevoca( revocaBancariaDTO: RevocaBancariaDTO,idUtente: number,idProgetto: number, idModalitaAgevolazione: number): Observable <boolean> {
      
      let url = this.configService.getApiURL() + '/restfacade/revoca/salvaRevoca';
      let params = new HttpParams()
      .set("idUtente", idUtente.toString())
      .set("idProgetto", idProgetto.toString())
      .set("idModalitaAgevolazione", idModalitaAgevolazione.toString());
      return this.http.post<boolean>(url, revocaBancariaDTO, { params: params });
    }

    /* salvaRevoca(revocaBancariaDTO,allegati, idProgetto,idModalitaAgevolazione): Observable<any> {
      console.log(revocaBancariaDTO);
      
      let url = this.configService.getApiURL() + '/restfacade/revoca/salvaRevoca';
      let params = new HttpParams()
      let formData = new FormData();
      for (var i = 0; i < allegati.length; i++) {
        console.log(allegati[i]);
        console.log(allegati[i].name);
        formData.append("file", allegati[i], allegati[i].name);
      }
      formData.append(
        "revocaBancariaDTO",
        JSON.stringify(revocaBancariaDTO)
      );
    
      formData.append("idModalitaAgevolazione", idModalitaAgevolazione); 
      formData.append("idProgetto", idProgetto);
      return this.http.post<any>(url, formData, { params: params });
    }  */

    getStorico (idUtente: number, idProgetto: number, idModalitaAgevolazione:number): Observable <Array<StoricoRevocaDTO>>{
      let url = this.configService.getApiURL() + '/restfacade/revoca/getStorico';
      let params = new HttpParams().set("idUtente", idUtente.toString()).set("idProgetto",idProgetto.toString())
      .set("idModalitaAgevolazione", idModalitaAgevolazione.toString());
      return this.http.get<Array<StoricoRevocaDTO>>(url, { params: params });
    }


    constructor(
        private configService: ConfigService,
        private http: HttpClient,
        private handleExceptionService: HandleExceptionService,

    ) { }

    private revocaBancariaDTO: RevocaBancariaDTO;
}

File: pbgestfinbo/pbgestfinbo-jar/src/main/java/it/csi/pbandi/pbgestfinbo/business/service/RevocaBancariaService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbgestfinbo.business.service;



import java.security.InvalidParameterException;
import java.util.ArrayList;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.Consumes;
import javax.ws.rs.DefaultValue;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import it.csi.pbandi.pbgestfinbo.dto.profilazione.RevocaBancariaDTO;
import it.csi.pbandi.pbgestfinbo.dto.profilazione.StoricoRevocaDTO;
import it.csi.pbandi.pbgestfinbo.exception.DaoException;

@Path("/revoca")
public interface RevocaBancariaService {
	
	@GET
	@Path("/getRevocheBancaria")
	@Produces(MediaType.APPLICATION_JSON)
	Response getRevocheBancarie (
			@DefaultValue("0") @QueryParam("idUtente") Long idUtente, 
			@DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
	        @Context HttpServletRequest req) throws DaoException;
	@GET
	@Path("/getRevocaBancaria")
	@Produces(MediaType.APPLICATION_JSON)
	RevocaBancariaDTO getRevocaBancaria (
			@DefaultValue("0") @QueryParam("idUtente") Long idUtente, 
			@DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
	        @Context HttpServletRequest req) throws DaoException;
	@POST
	@Path("/salvaRevoca")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	Response salvaRevocaBancaria(
			RevocaBancariaDTO revocaBancaria,
			@DefaultValue("0") @QueryParam("idUtente") long idUtente,
			@DefaultValue("0") @QueryParam("idProgetto") long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
			@Context HttpServletRequest request) 
			throws InvalidParameterException, Exception;
	
	@GET
	@Path("/getStorico")
	@Produces(MediaType.APPLICATION_JSON)
	ArrayList<StoricoRevocaDTO> getStorico(
			@DefaultValue("0") @QueryParam("idUtente") Long idUtente, 
			@DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
	        @Context HttpServletRequest req) throws DaoException;

}
